src/background/extension.ts

```diff
@@ -181,37 +212,82 @@ export class Extension {
     }
 
     async onCommand(command: string, url: string) {
-        if (this.startBarrier.isPending()) {
-            await this.startBarrier.entry();
+        /*
+         * We need the following line to re-use this function for
+         * context menu handlers.
+         */
+        const extension = this || globalThis.extension;
```

Comment:
What happens to "this" here? Is it "null"? Maybe try function assignment?
```typescript
onCommand = async (command: string, url: string) => {
    // ...
};
```
This way it will be bound to Extension instance when called as a function untied from an instance.